qt/aqt/clayout.py

```diff
@@ -159,10 +159,40 @@ def _fieldsOnTemplate(self, fmt):
             s += "+..."
         return s
 
+    def nextTab(self):
+        """ Fake implementation for MRU which just switches between the last 2 views
+        For a full implementation example, see: https://github.com/spyder-ide/spyder/pull/4302/files
+        """
+        if self.current_editor_index == 0:
+            if self.last_editor_index == 0:
+                self.tform.back_button.setChecked(True)
```

Comment:
Is this complexity necessary? Won't current_editor_index == last_editor_index only be the case before the user has switched between any tabs?

Reply:
No, if the user pressed `Ctrl+1` or clicked in the first tag twice in a row, the last index will be equal to the current index. As we are hardcoding this small "memory", it is simple to just list all cases because if someday we actually replace it by a full MRU algorithm, it should be simpler.

Reply:
Couldn't that be addressed with less code by not changing last_editor_index when the user clicks on the same tab again?

Reply:
I would not call these lines code, but a combination matrix. If we added that if on `last_editor_index`, we could remove one `inner` combination from each `outter` combination, but if somehow our `last_editor_index if` fail someday or we start changing the `last_editor_index` somewhere else and forget to add an if `last_editor_index` the code would break. As it is now, with the full matrix combination set, the code is more robust and less prone to fail (less fragile). Keeping the codebase more robust/prepared to failures should be better than removing a few lines and by adding an if somewhere else.

Reply:
IMHO listing out all the possibilities as you've done is more prone to accidental mistakes, and it makes it harder to maintain the code moving forward. I'm also still a bit skeptical as to the demand for this feature - I think the best option is to leave this PR open for a while and see if other people want to add their 2c.

Reply:
Closing for now; let's see if demand presents itself in the future